static/app/views/profiling/flamegraph.tsx

```diff
@@ -71,11 +79,37 @@ interface FlamegraphViewProps {
 function FlamegraphView(props: FlamegraphViewProps): React.ReactElement {
   const api = useApi();
   const organization = useOrganization();
+  const location = useLocation();
 
   const [requestState, setRequestState] = useState<RequestState<ProfileGroup>>({
     type: 'initial',
   });
 
+  const initialFlamegraphPreferencesState = useMemo((): DeepPartial<FlamegraphState> => {
+    if (requestState.type !== 'resolved') {
+      return decodeFlamegraphStateFromQueryParams(location.query);
+    }
+
+    const decodedState = decodeFlamegraphStateFromQueryParams(location.query);
+
+    return {
+      ...decodedState,
+      profiles: {
+        // We either initialize from query param or from the response
+        activeProfileIndex:
+          decodedState?.profiles?.activeProfileIndex ??
+          requestState.data.activeProfileIndex ??
+          0,
+      },
+    };
+  }, [requestState, location.query]);
+
+  // Sometimes the page we are coming from had scrollTop > 0, so we need to
+  // force restore it to top of page because scrolling on a flamegraph does not scroll the page
+  useEffect(() => {
+    document.scrollingElement?.scrollTo(0, 0);
+  }, []);
```

Comment:
Do we still need this? The page no longer scrolls at all now that the footer is gone.